azkaban-common/src/main/java/azkaban/executor/container/watch/FlowStatusManagerListener.java

```diff
@@ -130,7 +121,23 @@ private void validateAndPreProcess(AzPodStatusMetadata event) {
     }
     long eventCount = flowContainerEventCount.incrementAndGet();
     logEventCacheStats(eventCount);
-    validateTransition(event);
+
+    AzPodStatus currentStatus = getCurrentAzPodStatus(event.getPodName());
+    boolean isTransitionValid = validateTransition(event, currentStatus);
+    if (!isTransitionValid) {
+      IllegalStateException transitionException = new IllegalStateException(
+          format("Pod status transition is not supported %s -> %s, for pod %s",
+              currentStatus,
+              event.getAzPodStatus(),
+              event.getPodName()));
+      logger.error("Unsupported state transition.", transitionException);
+      try {
+        finalizeFlowAndDeleteContainer(event);
```

Comment:
Can you please delete container only for set of invalid transition? We might get pod down due to some statuses missing in state machine we have missed but not be deleting container for that.

Reply:
Yes, the deletion  will only happen for the invalid transitions. The logic underlying the `validateTransition` method will take care of ensuring this.